sparse/src/KokkosSparse_CcsMatrix.hpp

```diff
@@ -0,0 +1,290 @@
+template <class ScalarType, class OrdinalType, class Device,
+          class MemoryTraits = void,
+          class SizeType     = typename Kokkos::ViewTraits<OrdinalType*, Device,
+                                                       void, void>::size_type>
+class CcsMatrix {
+  static_assert(
+      std::is_signed<OrdinalType>::value,
+      "CcsMatrix requires that OrdinalType is a signed integer type.");
+
+ public:
+  //! Type of the matrix's execution space.
+  typedef typename Device::execution_space execution_space;
+  //! Type of the matrix's memory space.
+  typedef typename Device::memory_space memory_space;
+  //! Canonical device type
+  typedef Kokkos::Device<execution_space, memory_space> device_type;
+  typedef MemoryTraits memory_traits;
+
+  /// \brief Type of each entry of the "column map."
+  ///
+  /// The "column map" corresponds to the \c ptr array of column offsets in
+  /// compressed sparse column (CSR) storage.
+  typedef SizeType size_type;
+  //! Type of each value in the matrix.
+  typedef ScalarType value_type;
+  //! Type of each (column) index in the matrix.
+  typedef OrdinalType ordinal_type;
+  //! Type of the graph structure of the sparse matrix - consistent with Kokkos.
+  typedef Kokkos::StaticCcsGraph<ordinal_type, default_layout, device_type,
+                                 memory_traits, size_type>
+      staticccsgraph_type;
+  //! Type of the "column map" (which contains the offset for each column's
+  //! data).
+  typedef typename staticccsgraph_type::col_map_type col_map_type;
+  typedef Kokkos::View<value_type*, Kokkos::LayoutRight, device_type,
+                       MemoryTraits>
+      values_type;
+  //! Type of column indices in the sparse matrix.
+  typedef typename staticccsgraph_type::entries_type index_type;
+
+  /// \name Storage of the actual sparsity structure and values.
+  ///
+  /// CcsMatrix uses the compressed sparse column (CSC) storage format to
+  /// store the sparse matrix.
+  //@{
+  //! The graph (sparsity structure) of the sparse matrix.
+  staticccsgraph_type graph;
+  //! The 1-D array of values of the sparse matrix.
+  values_type values;
+  //@}
+
+ private:
+  /// \brief The number of rows in the CSC matrix
+  ordinal_type numRows_;
+
+ public:
+  /// \brief Default constructor; constructs an empty sparse matrix.
+  KOKKOS_INLINE_FUNCTION
+  CcsMatrix() : numRows_(0) {}
+
+  /// \brief Constructor that accepts a column map, row indices, and
+  ///   values.
+  ///
+  /// The matrix will store and use the column map, indices, and values
+  /// directly (by view, not by deep copy).
+  ///
+  /// \param label [in] The sparse matrix's label.
+  /// \param nrows [in] The number of rows.
+  /// \param ncols [in] The number of columns.
+  /// \param annz [in] The number of entries.
+  /// \param vals [in] The entries.
+  /// \param colmap [in] The column map (containing the offsets to the data in
+  /// each column). \param rows [in] The row indices.
```

Comment:
```suggestion
  /// each column).
  \param rows [in] The row indices.
```

Reply:
I'm not sure if doxygen will like having ``\param rows`` not on its own line

Reply:
I tried committing these small changes but github won't let me

Reply:
I think I have the branches on my fork protected